src/minizip/zip.c.diff

```diff
@@ -1,97 +1,15 @@
---- zip.c.orig	2014-06-10 18:55:32.778807408 +0200
-+++ zip.c	2014-06-11 20:27:18.355848501 +0200
-@@ -186,7 +186,7 @@
- #include "crypt.h"
- #endif
- 
--local linkedlist_datablock_internal* allocate_new_datablock()
-+local linkedlist_datablock_internal* allocate_new_datablock(void)
- {
-     linkedlist_datablock_internal* ldi;
-     ldi = (linkedlist_datablock_internal*)
-@@ -238,7 +238,7 @@
-     }
- 
-     ldi = ll->last_block;
--    from_copy = (unsigned char*)buf;
-+    from_copy = (const unsigned char*)buf;
- 
-     while (len>0)
-     {
-@@ -637,7 +637,7 @@
-   return relativeOffset;
- }
- 
--int LoadCentralDirectoryRecord(zip64_internal* pziinit)
-+local int LoadCentralDirectoryRecord(zip64_internal* pziinit)
- {
-   int err=ZIP_OK;
-   ZPOS64_T byte_before_the_zipfile;/* byte before the zipfile, (>0 for sfx)*/
-@@ -955,7 +955,7 @@
-     return zipOpen3(pathname,append,NULL,NULL);
- }
- 
--int Write_LocalFileHeader(zip64_internal* zi, const char* filename, uInt size_extrafield_local, const void* extrafield_local)
-+local int Write_LocalFileHeader(zip64_internal* zi, const char* filename, uInt size_extrafield_local, const void* extrafield_local)
- {
-   /* write the local header */
-   int err;
-@@ -1416,7 +1416,7 @@
- #ifdef HAVE_BZIP2
-     if(zi->ci.method == Z_BZIP2ED && (!zi->ci.raw))
-     {
--      zi->ci.bstream.next_in = (void*)buf;
-+      zi->ci.bstream.next_in = buf;
-       zi->ci.bstream.avail_in = len;
-       err = BZ_RUN_OK;
- 
-@@ -1450,7 +1450,7 @@
+--- zip.c.orig	2023-11-17 16:21:50.363864777 +0000
++++ zip.c	2023-11-17 16:26:04.671858368 +0000
+@@ -1401,7 +1401,7 @@
      else
  #endif
      {
--      zi->ci.stream.next_in = (Bytef*)buf;
+-      zi->ci.stream.next_in = (Bytef*)(uintptr_t)buf;
 +      zi->ci.stream.next_in = buf;
```

Comment:
@xroche viewing a diff-of-a-diff can be confusing, but I'd like to explain and highlight this change -- it was the only case where I was uncertain about how to resolve a merge conflict during the upgrade.

Basically: an additional typecast was added to the `minizip` sources here, and I've confirmed that that was intentional.

It caused this part of the diff/patch to not apply cleanly, and the resolution I chose for the pull request was to leave the assignment of `next_in = buf` as-is, without any typecasts.

However: I'm not totally confident about it.  It's a small detail, but I wanted to mention it anyway.